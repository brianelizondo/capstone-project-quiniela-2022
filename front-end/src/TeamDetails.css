.TeamDetails-logo{
    text-align: right;
    padding-right: 10px;
}
.TeamDetails-logo img{
    max-height: 200px !important;
    filter: drop-shadow(3px 3px 5px #000);
}
.TeamDetails-name{
    color: white;
    font-weight: 700;
    font-size: 50px;
    line-height: 70px;
    padding-top: 80px;
}

.TeamDetails-subtitle{
    color: white;
    font-weight: 100 !important;
    font-size: 18px;
    background-color: rgba(0,0,0, .60);
    display:inline-block;
    padding: 15px 15px;
    border-left: 5px solid black;
    margin-top: 50px;
    margin-bottom: 20px;
}

.TeamDetails-card{
    filter: drop-shadow(5px 5px 5px #000);
    margin-bottom: 50px;
    padding: 20px 20px 0px 20px;
}

.TeamDetails-stats-matchesplayed{
    font-family: 'Rubik', sans-serif;
    text-align: center;
    font-size: 60px;
    font-weight: 700;
    line-height: 60px;
    padding-top: 15px;
}
.TeamDetails-stats-text{
    color: gray;
    text-align: center;
    text-transform: capitalize;
    font-size: 14px;
}
.TeamDetails-stats-matcheswon, .TeamDetails-stats-matchesdrawn, .TeamDetails-stats-matcheslost{
    color: gray;
    text-transform: capitalize;
    text-align: center;
    font-size: 14px;
    line-height: 20px;
    margin-bottom: 15px;
}
.TeamDetails-stats-matcheswon span, .TeamDetails-stats-matchesdrawn span, .TeamDetails-stats-matcheslost span{
    font-family: 'Rubik', sans-serif;
    font-size: 20px;
    font-weight: 400;
    color: black;
    padding-right: 5px;
}
.TeamDetails-stats-matcheswon{
    border-bottom: 2px solid DarkGreen;
}
.TeamDetails-stats-matchesdrawn{
    border-bottom: 2px solid Goldenrod;
}
.TeamDetails-stats-matcheslost{
    border-bottom: 2px solid Crimson;
}

.TeamDetails-stats-various{
    padding-left: 40px;
}
.TeamDetails-stats-number{
    font-family: 'Rubik', sans-serif;
    font-size: 40px;
    font-weight: 400;
    color: black;
    padding-right: 5px;
    line-height: 38px;
}
.TeamDetails-stats-subtitle{
    font-size: 15px;
    text-transform: capitalize;
    color: gray;
    line-height: 18px;
}
.TeamDetails-stats-comment{
    font-size: 11px;
    color: darkgray;
    padding-bottom: 30px;
}

.TeamDetails-squad{
    float: left;
    margin: 0px 35px 60px 0px;
}
.TeamDetails-squad-name{
    font-size: 18px;
    font-weight: 700;
    color: black;
    text-align: center;
    line-height: 20px;
}
.TeamDetails-squad-pos{
    font-size: 13px;
    color: gray;
    text-align: center;
}