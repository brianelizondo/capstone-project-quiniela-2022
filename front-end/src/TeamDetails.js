import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { Card, ProgressBar, Image } from 'react-bootstrap';
import Loading from './Loading';

// import APIFootball api
import APIFootball from "./api-football";

function TeamDetails(){
    const { shortname } = useParams();
    // initial state
    const [loading, setLoading] = useState(false);
    const [team, setTeam] = useState({});
    const [teamStats, setTeamStats] = useState([]);
    const [teamSquad, setTeamSquads] = useState([]);

    useEffect(() => {
        setLoading(true);
        async function getTeam(){
            const resp = await APIFootball.getTeam(shortname);
            const respStats = await APIFootball.getTeamStats(shortname);
            const respSquad = await APIFootball.getTeamSquad(shortname);
            setTeam(resp);
            setTeamStats(respStats);
            setTeamSquads(respSquad);
        }
        getTeam();
        setLoading(false);
    }, [shortname]);

    const calculatePorc = (value, total) => {
        return ((value * 100) / total).toFixed(2);
    }

    if(loading){
        return <Loading />;
    }

    return (
        <div className="TeamDetails col-md-8 offset-md-2">
            <h1>Team in FIFA World Cup 2022</h1>
            <p>Details for the team "{team.name}"</p>
            
            <div className="TeamDetails-title">
                <div>{ team.apiInfo ? <img src={ team.apiInfo.logo } alt={ team.name } /> : "" }</div>
                <div>{ team.name }</div>
            </div>

            <div className="TeamDetails-standings">
                { teamStats.map( stats => 
                <Card key={ team.shortName }>
                    <Card.Title>Team Standings</Card.Title>
                    <Card.Body>
                        <div>
                            <div>{ stats.matches.played } matches played</div>    
                            <div>{ stats.matches.won } won</div>
                            <div>{ stats.matches.drawn } drawn</div>
                            <div>{ stats.matches.lost } lost</div>
                            <div>
                                <ProgressBar>
                                    <ProgressBar striped now={ calculatePorc(stats.matches.won, stats.matches.played) } key={1} />
                                    <ProgressBar striped variant="warning" now={ calculatePorc(stats.matches.drawn, stats.matches.played) } key={2} />
                                    <ProgressBar striped variant="danger" now={ calculatePorc(stats.matches.lost, stats.matches.played) } key={3} />
                                </ProgressBar>
                            </div>
                        </div>
                        <div>
                            <div>{ stats.goals.for }</div>    
                            <div>Goals</div>
                            <div>{ stats.goals.forAvg } avg. per match</div>
                        </div>
                        <div>
                            <div>{ stats.goals.against }</div>    
                            <div>Goals conceded</div>
                            <div>{ stats.goals.againstAvg } avg. per match</div>
                        </div>
                        <div>
                            <div>{ stats.cleanSheet.total }</div>    
                            <div>clean sheet</div>
                            <div>{ stats.cleanSheet.avg } avg. per match</div>
                        </div>
                        <div>
                            <div>{ stats.cards.yellow }</div>    
                            <div>yellow cards</div>
                            <div>{ stats.cards.yellowCAg } avg. per match</div>
                        </div>
                        <div>
                            <div>{ stats.cards.red }</div>    
                            <div>red cards</div>
                            <div>{ stats.cards.redCAg } avg. per match</div>
                        </div>
                    </Card.Body>
                </Card>
                )}
            </div>

            <div className="TeamDetails-squad">
                <Card>
                <Card.Title>Team Squad</Card.Title>
                    <Card.Body>
                        { teamSquad.map((player, idx) => 
                        <div key={idx}>
                            <div><Image src={ `https://media.api-sports.io/football/players/${player.id}.png` } alt={ player.name } roundedCircle fluid /></div>    
                            <div>{ player.name }</div>
                            <div>{ player.position }</div>
                        </div>
                        )}
                    </Card.Body>
                </Card>
            </div>
        </div>
    );
}

export default TeamDetails;