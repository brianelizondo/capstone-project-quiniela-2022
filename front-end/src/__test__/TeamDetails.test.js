import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MemoryRouter } from "react-router-dom";

import TeamDetails from '../TeamDetails';

// Smoke Test
describe("Smoke Test", () => {
    it("renders 'TeamDetails' component without crashing", () => {
        render(
            <MemoryRouter initialEntries={["/teams/ARG"]}>
                <TeamDetails />
            </MemoryRouter>
        );
        const textFound = screen.getByText("Team Squad");
        expect(textFound).toBeInTheDocument();
    });
});

// Snapshot Tests
describe("Snapshot Test", () => {
    it("'TeamDetails' matches snapshot", () => {
        const {asFragment} = render(
            <MemoryRouter initialEntries={["/teams/ARG"]}>
                <TeamDetails />
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
