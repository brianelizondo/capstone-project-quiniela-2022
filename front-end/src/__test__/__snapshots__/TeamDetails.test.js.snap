// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test 'TeamDetails' matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TeamDetails col-md-8 offset-md-2"
  >
    <h1>
      Team in FIFA World Cup 2022
    </h1>
    <p>
      Details for the team ""
    </p>
    <div
      class="TeamDetails-title"
    >
      <div />
      <div />
    </div>
    <div
      class="TeamDetails-standings"
    />
    <div
      class="TeamDetails-squad"
    >
      <div
        class="card"
      >
        <div
          class="card-title h5"
        >
          Team Squad
        </div>
        <div
          class="card-body"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
